fix(nimbus): make fenix enrollment integration test wait deterministically for log-state output

The Fenix enrollment integration test ([test_fenix_enrollment.py](experimenter/tests/integration/nimbus/android/test_fenix_enrollment.py)) is intermittently failing on the **beta** channel job with:

```
AssertionError: No log-state row found for fenix-beta-integration-test.
```

Three recent failures, all on beta, release passing each time:

- https://github.com/mozilla/experimenter/actions/runs/25881965585/job/76063971959
- https://github.com/mozilla/experimenter/actions/runs/25873884991
- https://github.com/mozilla/experimenter/actions/runs/25871997491

## Cause

The test does two `am start` intents (via `nimbus-cli`) and uses fixed `time.sleep()` calls between them:

```python
nimbus-cli ... enroll ...
time.sleep(APP_APPLY_WAIT)   # 15s
nimbus-cli ... log-state
time.sleep(LOG_STATE_WAIT)   # 5s
logcat = adb logcat -d
```

In the failing runs, logcat ends right at the `log-state` table header (`Slug | Features | Branch`) — the SDK hadn't yet emitted the per-experiment data row when we captured. Fenix beta boots slower than release in the emulator, so the 5s window after `log-state` races the SDK's async row print.

## Fix

Replace both fixed sleeps with a bounded poll on `adb logcat -d`:

- After `nimbus-cli enroll`, wait until a known post-apply `nimbus_client` marker appears (or timeout).
- After `nimbus-cli log-state`, wait until the experiment-row regex matches (or timeout).

This both removes the flake and shortens the happy-path runtime (no need to over-wait).

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-6891)

--- a/experimenter/tests/integration/nimbus/android/test_fenix_enrollment.py
+++ b/experimenter/tests/integration/nimbus/android/test_fenix_enrollment.py
@@ -9,7 +9,8 @@ from nimbus.models.base_dataclass import BaseExperimentApplications
 
 FENIX_APP = BaseExperimentApplications.FIREFOX_FENIX.value
 APP_APPLY_WAIT = 15
-LOG_STATE_WAIT = 5
+LOG_STATE_TIMEOUT = 60
+LOG_STATE_POLL_INTERVAL = 1
 
 
 @pytest.mark.fenix_enrollment
@@ -59,17 +60,24 @@ def test_fenix_enrollment(
     subprocess.check_call(
         ["nimbus-cli", "--app", FENIX_APP, "--channel", fenix_channel, "log-state"]
     )
-    time.sleep(LOG_STATE_WAIT)
-
-    logcat = subprocess.check_output(["adb", "logcat", "-d"], text=True)
 
     pattern = re.compile(
         rf"nimbus_client:\s*{re.escape(experiment_slug)}\s+\|\s*\S+\s+\|\s*(\S+)"
     )
-    match = pattern.search(logcat)
+
+    logcat = ""
+    match = None
+    deadline = time.monotonic() + LOG_STATE_TIMEOUT
+    while time.monotonic() < deadline:
+        logcat = subprocess.check_output(["adb", "logcat", "-d"], text=True)
+        match = pattern.search(logcat)
+        if match is not None:
+            break
+        time.sleep(LOG_STATE_POLL_INTERVAL)
+
     nimbus_lines = [line for line in logcat.splitlines() if "nimbus_client" in line]
     assert match is not None, (
-        f"No log-state row found for {experiment_slug}.\n"
+        f"No log-state row found for {experiment_slug} after {LOG_STATE_TIMEOUT}s.\n"
         f"--- last 30 nimbus_client lines ---\n" + "\n".join(nimbus_lines[-30:])
     )
     enrolled_branch = match.group(1)

